integration_tests/src/lib.rs

```diff
@@ -2662,6 +3052,142 @@ mod test {
+            .add_subvolume_create_guest(
+                test_downstairs_set.opts(),
+                volume::RegionExtentInfo {
+                    block_size: BLOCK_SIZE as u64,
+                    blocks_per_extent: test_downstairs_set.blocks_per_extent(),
+                    extent_count: test_downstairs_set.extent_count(),
+                },
+                1,
+                None,
+            )
+            .await?;
+
+        volume.activate().await?;
+
+        let random_buffer = {
+            let mut random_buffer =
+                vec![0u8; volume.total_size().await? as usize];
+            rand::thread_rng().fill(&mut random_buffer[..]);
+            random_buffer
+        };
+
+        volume
+            .write(
+                Block::new(0, BLOCK_SIZE.trailing_zeros()),
+                Bytes::from(random_buffer.clone()),
+            )
+            .await?;
+
+        volume.deactivate().await.unwrap();
+
+        // Restart the three downstairs in read only mode.
+        test_downstairs_set.reboot_read_only().await?;
+
+        // Restart our volume with the restarted read-only downstairs.
+        let mut volume = Volume::new(BLOCK_SIZE as u64, csl());
+        volume
+            .add_subvolume_create_guest(
+                test_downstairs_set.opts(),
+                volume::RegionExtentInfo {
+                    block_size: BLOCK_SIZE as u64,
+                    blocks_per_extent: test_downstairs_set.blocks_per_extent(),
+                    extent_count: test_downstairs_set.extent_count(),
+                },
+                2,
+                None,
+            )
+            .await?;
+
+        volume.activate().await?;
+
+        // Make the new downstairs, but don't start it
+        let mut new_ds = TestDownstairs::new(
+            "127.0.0.1".parse().unwrap(),
+            true,
+            true,
+            5,
+            2,
+            false,
+            Backend::RawFile,
+        )
+        .await
+        .unwrap();
+
+        let clone_source =
+            test_downstairs_set.downstairs1.repair_address().await;
+
+        // Clone a read only downstairs into the new downstairs region.
+        new_ds.reboot_clone(clone_source).await.unwrap();
+        // Start the new downstairs read only.
+        new_ds.reboot_read_only().await.unwrap();
+
+        // Replace a downstairs in our RO volume with the new downstairs we
+        // just cloned.
+        let res = volume
+            .replace_downstairs(
+                test_downstairs_set.opts().id,
+                test_downstairs_set.downstairs1.address().await,
+                new_ds.address().await,
+            )
+            .await
+            .unwrap();
+
+        assert_eq!(res, ReplaceResult::Started);
+
+        // We can use the result from calling replace_downstairs to
+        // intuit status on progress of the replacement.
+        loop {
+            tokio::time::sleep(tokio::time::Duration::from_secs(2)).await;
```

Comment:
It will do the replacement of one downstairs target with another.  But, since it's a RO volume, we don't
try to do any LiveRepair. I'm sure it would be less than a second to do the replacement, but I did
not want to spam the logs with as many times as we would check for it being done in that second.